src/cli.rs

```diff
@@ -158,6 +162,23 @@ pub fn list_tasks(
         status_filter && task.check_metadata(metadata.iter())
       }
     })
+    .filter(|(_, task)| {
+      if !name_filter.is_empty() {
+        let mut name_filter = name_filter.clone();
+
+        for word in task.name().split_ascii_whitespace() {
+          let word_found = name_filter.remove(word);
+
+          if word_found && name_filter.is_empty() {
+            break;
+          }
+        }
+
+        name_filter.is_empty()
```

Comment:
I think you can simply replace all this with 
```suggestion
        for word in task.name().split_ascii_whitespace() { 
          let _ = name_filter.remove(word);
         
          if name_filter.is_empty() {
            return true;
          }
        }
        
        return false;
```

The fact that the word is found is not really important: what is important is the state of the `HashSet`. If it’s empty, it means that we have found everything and we can short-circuit and leave. Given that, if the control flow exits the `for`, then we have still something in the `HashSet` and we then the test is failed.

Reply:
The early return is fine but the `word_found` is actually a small optimization because if [remove](https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.remove) returns false the set isn't modified and we do not need to check the set. But it may not matter because isEmpty should be inlined and a simple value access.

Reply:
The early return part is pushed.